src/aux-records/RecordsServer.ts

```diff
@@ -2090,6 +2140,178 @@ export class RecordsServer {
+                                contractedUserId:
+                                    GetXpUserById.optional().nullable(),
+                            })
+                            .refine(
+                                (contract) => {
+                                    if (contract.status === 'open') {
+                                        //* Open contracts must have a contracted user
+                                        return (
+                                            contract.contractedUserId ??
+                                            undefined !== undefined
+                                        );
+                                    } else if (contract.status === 'draft') {
+                                        /**
+                                         * * Draft contracts from the database's perspective should not be expecting a contracted user yet.
+                                         * * This is because the contracted user is only set when the contract is opened.
+                                         */
+                                        return (
+                                            contract.contractedUserId ??
+                                            undefined === undefined
+                                        );
+                                    }
+                                },
+                                {
+                                    message:
+                                        'Contract must contain contractedUserId (only when status is "open").',
+                                    path: ['contractedUserId'],
+                                }
+                            )
+                            .refine(
+                                (contract) => {
+                                    if (contract.status === 'open') {
+                                        return (
+                                            contract.accountCurrency ??
+                                            undefined !== undefined
+                                        );
+                                    } else if (contract.status === 'draft') {
+                                        return (
+                                            contract.accountCurrency ??
+                                            undefined === undefined
+                                        );
+                                    }
+                                },
+                                {
+                                    message:
+                                        'Contract must contain accountCurrency (only when status is "open").',
+                                    path: ['accountCurrency'],
+                                }
+                            ),
+                    })
+                )
+                .handler(async ({ contract }, context) => {
+                    const authUser = await this._validateSessionKey(
```

Comment:
@LandonSiler This procedure should check to see if the xpController is supported as well.